mergekit/metric_methods/all_metrics.py

```diff
@@ -0,0 +1,373 @@
+        res = {}
+        models = list(q_proj.keys())
+
+        k_proj_0, v_proj_0, q_proj_0, o_proj_0 = group_attn_head_weights(k_proj[models[0]], q_proj[models[0]], v_proj[models[0]], o_proj[models[0]], self.weight_info)
+        k_proj_1, v_proj_1, q_proj_1, o_proj_1 = group_attn_head_weights(k_proj[models[1]], q_proj[models[1]], v_proj[models[1]], o_proj[models[1]], self.weight_info)
+        
+        # Metrics for K, V, Q, O projections
+
+        model_0_heads = torch.cat([k_proj_0, v_proj_0, q_proj_0, o_proj_0], dim=1)
+        model_1_heads = torch.cat([k_proj_1, v_proj_1, q_proj_1, o_proj_1], dim=1)
+        
+        # Metrics for heads
+        res.update(mse([model_0_heads.view(model_0_heads.shape[0], -1), 
+                        model_1_heads.view(model_1_heads.shape[0], -1)], 
+                        return_heatmap=True))
+        res.update(cossim([model_0_heads.view(model_0_heads.shape[0], -1), 
+                           model_1_heads.view(model_1_heads.shape[0], -1)], 
+                           return_heatmap=True))
+        res.update(scale([model_0_heads.view(model_0_heads.shape[0], -1),
+                            model_1_heads.view(model_1_heads.shape[0], -1)],
+                            return_heatmap=True))
+        res.update(SMAPE([model_0_heads.view(model_0_heads.shape[0], -1),
+                            model_1_heads.view(model_1_heads.shape[0], -1)]))
+
+        return res
+
+    def group_label(self) -> Optional[str]:
+        # Use max of the group labels
+        return max([gather_tensor.group_label() for gather_tensor in list(self.weights.values())]) # Check this (X)
+    
+    def __hash__(self):
+        return hash(self.weight_info)
+
+    def __eq__(self, other):
+        if not isinstance(other, AttnTask):
+            return False
+        return self.weight_info == other.weight_info
+
+class DummyTask(Task[torch.Tensor]):
+    gather_tensors: GatherTensors
+    weight_info: WeightInfo
+    def uses_accelerator(self) -> bool:
+        return True
+
+    def arguments(self) -> Dict[str, Task]:
+        return {"tensors": self.gather_tensors}
+
+    def execute(
+        self, **_kwargs
+    ) -> torch.Tensor:
+        
+        return 
+    
+    def group_label(self) -> Optional[str]:
+        return self.gather_tensors.group_label()
+
+# Metric method
+   
+class AllMetric(MetricMethod):
+    attn_weight_tensors: Optional[list] = []
+    attn_weight_infos: Optional[list] = []
+
+    attn_weight_dict: Optional[Dict[str, torch.Tensor]] = {}
+    attn_info_dict: Optional[Dict[str, WeightInfo]] = {}
+
+    attn_parts: Optional[List[str]] = ['k_proj', 'v_proj', 'q_proj', 'o_proj'] # hard-coded for now
+    block_count: Optional[int] = 0
+    def make_task(
+        self,
+        *,
+        output_weight: WeightInfo,
+        parameters: Optional[Dict[str, Any]] = None,
+        tensors: GatherTensors,
+        **_kwargs,
+    ) -> Task:
+        
+        if 'self_attn' in output_weight.name:
```

Comment:
Down the line we probably want this split to be done based on new fields in ArchitectureInfo but this is good for now!